### PR TITLE
[proto] Speed improvement for autocontrast op

### PR DESCRIPTION
**Pull request by vfdev-5**
Results:

```
[---------------- Autocontrast cpu torch.uint8 ---------------]
                     |  autocontrast stable  |  autocontrast v2
1 threads: ----------------------------------------------------
      (3, 400, 400)  |          969          |        844      
6 threads: ----------------------------------------------------
      (3, 400, 400)  |          320          |        274      

Times are in microseconds (us).

[--------------- Autocontrast cuda torch.uint8 ---------------]
                     |  autocontrast stable  |  autocontrast v2
1 threads: ----------------------------------------------------
      (3, 400, 400)  |          200          |        150      
6 threads: ----------------------------------------------------
      (3, 400, 400)  |          201          |        149      

Times are in microseconds (us).

[--------------- Autocontrast cpu torch.float32 --------------]
                     |  autocontrast stable  |  autocontrast v2
1 threads: ----------------------------------------------------
      (3, 400, 400)  |          529          |        418      
6 threads: ----------------------------------------------------
      (3, 400, 400)  |          201          |        157      

Times are in microseconds (us).

[-------------- Autocontrast cuda torch.float32 --------------]
                     |  autocontrast stable  |  autocontrast v2
1 threads: ----------------------------------------------------
      (3, 400, 400)  |          166          |        115      
6 threads: ----------------------------------------------------
      (3, 400, 400)  |          165          |        114      

Times are in microseconds (us).
```

cc @datumbox @bjuncek @pmeier

### CODE REVIEW

File: torchvision/prototype/transforms/functional/_color.py [hunk 1]
```diff
@@ -211,7 +211,34 @@ def solarize(inpt: features.InputTypeJIT, threshold: float) -> features.InputTyp
...
+
+    if image.numel() == 0:
+        # exit earlier on empty images
+        return image
+
+    bound = 1.0 if image.is_floating_point() else 255.0
+    dtype = image.dtype if torch.is_floating_point(image) else torch.float32
+
+    minimum = image.amin(dim=(-2, -1), keepdim=True).to(dtype)
+    maximum = image.amax(dim=(-2, -1), keepdim=True).to(dtype)
```
Comment (vfdev-5) [inline torchvision/prototype/transforms/functional/_color.py:232]: We can use `torch.aminmax` here as it does not support tuple of dims. Reshaping the input does not improve runtime.

File: torchvision/prototype/transforms/functional/_color.py [hunk 2]
```diff
@@ -211,7 +211,34 @@ def solarize(inpt: features.InputTypeJIT, threshold: float) -> features.InputTyp
...
+
+    scale = bound / (maximum - minimum)
+    eq_idxs = maximum == minimum
```
Comment (datumbox) [inline torchvision/prototype/transforms/functional/_color.py:235]: Reading back to original code that I wrote, I see I've used `eq_idxs = torch.isfinite(scale).logical_not()` for the same thing. To be honest, I don't understand why I did that (it looks kind of stupid). Just to be safe, you tested this logic with special cases where the values are the same across different channels and the result is identical right?

From code I expect that your approach is faster and better. Just being a bit paranoid.

**Review by datumbox — APPROVED** [review]
LGTM, thanks! Just one question:

File: torchvision/prototype/transforms/functional/_color.py (hunk 2 above)
Comment (vfdev-5) [reply torchvision/prototype/transforms/functional/_color.py:235]: Thanks for asking, I'll check with more attention to that

Comment (vfdev-5) [reply torchvision/prototype/transforms/functional/_color.py:235]: I did few other checks and I think `eq_idxs = maximum == minimum` should be equivalent to `eq_idxs = torch.isfinite(scale).logical_not()`. So, we can keep PR's implementation.

**Pull request merged**